python/servo/bootstrap.py

```diff
@@ -13,26 +13,55 @@
 
 import servo.packages as packages
 from servo.util import extract, download_file, host_triple
+from mach_bootstrap import _get_virtualenv_script_dir
 
 
-def salt(context, force=False):
+def run_as_root(command):
+    if os.geteuid() != 0:
+        try:
+            # Check if sudo'ing possible
+            subprocess.check_call(['sudo', '-v'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            command.insert(0, 'sudo')
+        except:
+            command = ['su', 'root', '-c', ' '.join(command)]
```

Comment:
Where is `su` needed? I much prefer using `sudo` in all circumstances, since `su` 1) uses root's environment instead of the user's environment and 2) requires the root password instead of the user's password.